### PR TITLE
rbac: new command to list/delete credentials

### LINKED ISSUE #5750: Add the ability to remove credentials from AD Service Principal
#5642 Hello,

Could you please add the ability to remove credentials or expire credentials from AD Service principal.  It seems there is only support to add new credentials but not remove/expire old ones.

This logic does exist in powershell and not yet in Azure CLI.

-Thank you

---

### Environment summary

Install Method (e.g. pip, interactive script, apt-get, Docker, MSI, edge build) / CLI version (`az --version`) / OS version / Shell Type (e.g. bash, cmd.exe, Bash on Windows)  

```
```

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #5750 

### General Guidelines

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [x] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5811
This is an **experimental preview** for @microsoft users.

**Comment by yugangw-msft** [comment]
@troydai, could you please take a look at this one?

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -670,25 +670,86 @@ def show_service_principal(client, identifier):
...
+    sp_object_id = _resolve_service_principal(client.service_principals, identifier)
+    if cert:
+        result = list(client.service_principals.list_key_credentials(sp_object_id))
     else:
-        client.service_principals.delete(sp.object_id)
+        result = list(client.service_principals.list_password_credentials(sp_object_id))
+
+    to_delete = next((x for x in result if x.key_id == key_id), None)
+
+    # we will try to delete the creds at service principal level, if nout found, we try application level
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: nit: typo 'nout'

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py [hunk 2]
```diff
@@ -670,25 +670,86 @@ def show_service_principal(client, identifier):
...
+            else:
+                result = list(client.applications.list_password_credentials(app_object_id))
+            to_delete = next((x for x in result if x.key_id == key_id), None)
+            if to_delete:
+                result.remove(to_delete)
+                if cert:
+                    return client.applications.update_key_credentials(app_object_id, result)
+                return client.applications.update_password_credentials(app_object_id, result)
+
+    raise CLIError("'{}' does't exist in the service principal of '{}' or associated application".format(
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py]: typo `doesn't `

**Pull request merged**